homeassistant/components/fan/isy994.py

```diff
@@ -0,0 +1,126 @@
+
+_LOGGER = logging.getLogger(__name__)
+
+VALUE_TO_STATE = {
+    0: SPEED_OFF,
+    63: SPEED_LOW,
+    64: SPEED_LOW,
+    190: SPEED_MED,
+    191: SPEED_MED,
+    255: SPEED_HIGH,
+}
+
+STATE_TO_VALUE = {}
+for key in VALUE_TO_STATE:
+    STATE_TO_VALUE[VALUE_TO_STATE[key]] = key
+
+STATES = [SPEED_OFF, SPEED_LOW, SPEED_MED, SPEED_HIGH]
+
+
+def setup_platform(hass, config: ConfigType,
+                   add_devices: Callable[[list], None], discovery_info=None):
+    """Setup the ISY994 fan platform."""
+    if ISY is None or not ISY.connected:
+        _LOGGER.error('A connection has not been made to the ISY controller.')
+        return False
+
+    devices = []
+
+    for node in filter_nodes(NODES, states=STATES):
+        devices.append(ISYFanDevice(node))
+
+    for program in PROGRAMS.get(DOMAIN, []):
+        try:
+            status = program[KEY_STATUS]
+            actions = program[KEY_ACTIONS]
+            assert actions.dtype == 'program', 'Not a program'
+        except (KeyError, AssertionError):
+            pass
+        else:
+            devices.append(ISYFanProgram(program.name, status, actions))
+
+    add_devices(devices)
+
+
+class ISYFanDevice(ISYDevice, FanEntity):
+    """Representation of an ISY994 fan device."""
+
+    def __init__(self, node) -> None:
+        """Initialize the ISY994 fan device."""
+        ISYDevice.__init__(self, node)
+        self.speed = self.state
+
+    @property
+    def state(self) -> str:
+        """Get the state of the ISY994 fan device."""
+        return VALUE_TO_STATE.get(self.value, STATE_UNKNOWN)
+
+    def set_speed(self, speed: str) -> None:
+        """Send the set speed command to the ISY994 fan device."""
+        if not self._node.on(val=STATE_TO_VALUE.get(speed, 0)):
+            _LOGGER.debug('Unable to set fan speed')
+        else:
+            self.speed = self.state
+
+    def turn_on(self, speed: str=None, **kwargs) -> None:
+        """Send the turn on command to the ISY994 fan device."""
+        self.set_speed(speed)
+
+    def turn_off(self, **kwargs) -> None:
+        """Send the turn off command to the ISY994 fan device."""
+        if not self._node.off():
+            _LOGGER.debug('Unable to set fan speed')
+        else:
+            self.speed = self.state
+
+
+class ISYFanProgram(ISYFanDevice):
+    """Representation of an ISY994 fan program."""
+
+    def __init__(self, name: str, node, actions) -> None:
+        """Initialize the ISY994 fan program."""
+        ISYFanDevice.__init__(self, node)
+        self._name = name
+        self._actions = actions
+        self.speed = STATE_ON if self.is_on else STATE_OFF
+
+    @property
+    def is_on(self) -> bool:
+        """Get whether the ISY994 fan program is on."""
+        return bool(self.value)
+
+    @property
+    def state(self) -> str:
+        """Get the state of the ISY994 fan program."""
+        return STATE_ON if self.is_on else STATE_OFF
```

Comment:
Same here: `return STATE_ON if self.value else STATE_OFF` and then rely on the inherited `is_on`